monkey/monkey_island/cc/models/zero_trust/finding.py

```diff
@@ -48,11 +48,19 @@ def get_pillars(self):
 
     # Creation methods
     @staticmethod
-    def save_finding(test: str, status: str, detail_ref):
-        finding = Finding(test=test,
-                          status=status,
-                          details=detail_ref)
+    def save_finding(test: str,
+                     status: str,
+                     detail_ref: Union[MonkeyFindingDetails, ScoutSuiteFindingDetails]):
```

Comment:
Except there is nothing in common between `MonkeyFindingDetails` and `ScoutSuiteFindingDetails`. The best choice is probably to refactor `Finding` into `ScoutSuiteFinding` and `MonkeyFinding`.

Reply:
But in that case pulling all findings is no longer a trivial task

Reply:
You're right. Refactoring `Finding` into `ScoutSuiteFinding` and `MonkeyFinding` is the right thing to do.
Why pulling all findings won't be trivial after this change? If they both inherit from a generic `Finding` document, you can easily query all of them, or just 1 type of findings.

Reply:
There's a reason why `MonkeyFindingDetails` and `ScoutSuiteFindingDetails` are separate documents. Codebase has 2 separate usages for the data: pull only the data of `Finding` and pull full `Finding` (with details). If we essentially merge details data with finding data into a single document, by default you will pull all data when pulling `Finding`, which is a major performance issue. Either way, a single `Union` type hint doesn't justify full scale refactoring IMO.

Reply:
The `Union` isn't the reason to justify it, it's just a trigger that should make you question your design.

I'm not suggesting to merge the Finding and the FindingDetails documents into 1.
I'm suggesting you refactor the generic `Finding` document so that it'll be an abstract document. `ScoutSuiteFinding` & `MonkeyFinding` should inherit from it. The `Finding` document should have all the fields which are relevant to ALL findings. `ScoutSuiteFinding` & `MonkeyFinding` should have only the relevant fields relevant for them.
In case you have further details for each, you can have `ScoutSuiteFinding` contain a lazy reference to to a `ScoutSuiteFindingDetails` document, and have the same done for `MonkeyFinding`